source/game/system/KPadController.cc

```diff
@@ -236,6 +301,87 @@ u8 KPadGhostTrickButtonsStream::readVal() const {
     return currentSequence >> 0x8 & ~0x80;
 }
 
+/* ================================ *
+ *     HOST CONTROLLER
+ * ================================ */
+
+KPadHostController::KPadHostController() = default;
+
+KPadHostController::~KPadHostController() = default;
+
+ControlSource KPadHostController::controlSource() {
+    return ControlSource::Host;
+}
+
+void KPadHostController::reset(bool driftIsAuto) {
+    m_driftIsAuto = driftIsAuto;
+    m_raceInputState.reset();
+    m_connected = true;
+}
+
+/// @brief Sets the inputs of the controller.
```

Comment:
Well there's several other functions that ALSO set the inputs... So what's different about this one? Is it just because we're exposing several of these depending on how other developers may want to interface with Kinoko? I know the answer is yes, but I think it would be good to specify in the `@brief` that **that** is the reason why there are several ways to set inputs.

Reply:
There's nothing different - they all perform more or less identically, and it is up to developers to decide which one they want to use.